File: solutions/opus-1/opus/ruby/wordfreq.rb
counts = Hash.new(0)
$stdin.binmode.read.scan(/[A-Za-z]+/) { |w| counts[w.downcase] += 1 }
out = counts.sort_by { |w, c| [-c, w] }.map { |w, c| "#{w} #{c}\n" }.join
$stdout.write(out)

File: solutions/opus-1/opus/python/wordfreq.py
import re
import sys
from collections import Counter

data = sys.stdin.buffer.read().decode("latin-1")
counts = Counter(w.lower() for w in re.findall(r"[A-Za-z]+", data))
out = [f"{w} {c}" for w, c in sorted(counts.items(), key=lambda x: (-x[1], x[0]))]
sys.stdout.write("\n".join(out) + ("\n" if out else ""))
